editor/scene_tree_dock.cpp

```diff
@@ -2922,22 +2929,24 @@ void SceneTreeDock::_create() {
 		}
 
 		Node *parent = nullptr;
+		int original_position = -1;
 		if (only_one_top_node) {
 			parent = top_node->get_parent();
+			original_position = top_node->get_index();
```

Comment:
@RedMser This should be `original_position = top_node->get_index(false);`, as `move_child` takes in an index not accounting for internal nodes.

Here's an example of it failing for TabContainer (which has front-internal TabBar child):
![godot windows editor x86_64_IqOzGwV6GD](https://github.com/user-attachments/assets/e6f76b41-863a-4bb7-ba3b-aca954af46a3)
After `First` -> `Reparent to New Node...`:
![godot windows editor x86_64_NDmkEd7XBO](https://github.com/user-attachments/assets/6b77b116-e132-461b-9148-e4dbd94f6c07)
Correct result (currently only if parent has no internal children):
![godot windows editor x86_64_z8THw1XRG6](https://github.com/user-attachments/assets/1bfdcfc4-876b-49d4-91a7-6eae5a7f3cef)

Reply:
Ah I see. Would you like to create a followup PR to fix this, or should I do so?

Reply:
Go ahead! :slightly_smiling_face:

Reply:
Done, thanks for the info!
https://github.com/godotengine/godot/pull/96646